src/unit_tests/wazuh_db/test_wdb_agents_helpers.c

```diff
@@ -339,6 +339,180 @@ void test_wdb_agents_vuln_cve_clear_success(void **state)
     assert_int_equal(OS_SUCCESS, ret);
 }
 
+/* Tests wdb_agents_vuln_cve_update_status */
+
+test_wdb_agents_vuln_cve_update_status_error_json(void **state){
+    int ret = 0;
+    int id = 1;
+    const char *old_status = "valid";
+    const char *new_status = "obsolete";
+
+    will_return(__wrap_cJSON_CreateObject, NULL);
+
+    expect_string(__wrap__mdebug1, formatted_msg, "Error creating data JSON for Wazuh DB.");
+
+    ret = wdb_agents_vuln_cve_update_status(id, old_status, new_status, NULL);
+
+    assert_int_equal(OS_INVALID, ret);
+}
+
+test_wdb_agents_vuln_cve_update_status_error_socket(void **state){
+    int ret = 0;
+    int id = 1;
+    const char *old_status = "valid";
+    const char *new_status = "obsolete";
+
+    const char *json_str = strdup("{\"old_status\":\"valid\",\"new_status\":\"obsolete\"}");
```

Comment:
It's recommended to use os_strdup instead od strdup

Reply:
Fixed